Regular-Question-Answers/symmetric-tree.cpp

```diff
@@ -0,0 +1,70 @@
+// Author: Ceren Yaşar
+// Question Link: https://leetcode.com/problems/symmetric-tree/
+
+// Time Complexity: O(N)
+// Memory Complexity: O(N)
+
+// This solution uses breadth first search since we start comparisons from the closest child.
+
+/**
+ * Definition for a binary tree node.
+ * struct TreeNode {
+ *     int val;
+ *     TreeNode *left;
+ *     TreeNode *right;
+ *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
+ *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
+ *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
+ * };
+ */
+
+class Solution {
+public:
+    bool isSymmetric(TreeNode* root) {
+        
+        // This queue keeps pairs of tree nodes that are supposed to have equal values
+        queue<pair<TreeNode*, TreeNode*>> nextToCheck;
+        
+        // If the root node stands alone, it is symmetric
+        if (!root->left && !root->right)
+            return true;
+        // If the root node has only one child, it is asymmetric
+        else if (!root->left || !root->right)
+            return false;
+        
+        // First pair that is supposed to have equal values is direct children of the root
+        nextToCheck.push({root->left,root->right});
+        
+        TreeNode *a, *b;
+
+        while (!nextToCheck.empty()) {
+
+            a = nextToCheck.front().first;
+            b = nextToCheck.front().second;
+            nextToCheck.pop();
+            
+            // If a and b have equal values, we can look for next nodes to check which would be
+            // direct children of a's and b's.
+            if (a->val == b->val) {
+                // Symmetry axis is placed at the center, so one nodes left child should mirror other's right, vice versa.
+                // If one of the children mentioned above do not exist, the tree is asymmetric.
+                if (a->left && b->right)
+                    nextToCheck.push({a->left,b->right});
+
+                else if ( (a->left && !b->right) || (!a->left && b->right) )
+                    return false;
+
+                if (a->right && b->left)
+                    nextToCheck.push({a->right,b->left});
+
+                else if ( (a->right && !b->left) || (!a->right && b->left) )
+                    return false;
+            }
+            // If a and b have different values
+            else
```

Comment:
I have read that although it is shorter, this method is not usually recommended as it makes your code difficult to read. Maybe i could even use curly brackets, what do you think?

Reply:
In this case, if written in one line, the code is still clear and leaves less empty space. If it was something like `else (condition)`, the choice of writing in two lines would have been more plausible. I don't think that using curly brackets here is suitable either, because the code is final and you are not considering adding more instructions.
I already merged your request and edited that line by myself since we don't have time.